libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSDKCryptoUtils.m

```diff
@@ -499,4 +629,44 @@ + (size_t)encodeLength:(unsigned char *)buf length:(size_t)length {
     return i + 1;
 }
 
++ (nullable SecKeyRef)getECKeyRefWithTag:(NSString*)keyTagString {
+    NSData *keyTag = [keyTagString dataUsingEncoding:NSUTF8StringEncoding];
+    
+    NSDictionary* getquery = @{ (id)kSecClass:              (id)kSecClassKey,
+                                (id)kSecAttrApplicationTag: keyTag,
+                                (id)kSecReturnRef:          @YES };
+    
+    SecKeyRef keyRef = NULL;
+    OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)getquery, (CFTypeRef *)&keyRef);
+    
+    if (status != errSecSuccess && status != errSecItemNotFound) {
+        // Handle the error. . .
+        NSError *error = [NSError errorWithDomain:NSOSStatusErrorDomain code:status userInfo:nil];
+        [SFSDKCoreLogger e:[self class] format:@"Error getting EC SecKeyRef with tag %@. Error code: %@", keyTagString, error.localizedDescription];
+        return nil;
+    }
+    
+    return keyRef;
+}
+
++ (void)deleteKeyByTag:(NSString*)keyTagString {
```

Comment:
@wmathurin We should either return an error or a boolean to allow calling code to handle  the failure.